### ISSUE TITLE
Question: How to Maximize, Minimize, Restore WinUI window from code?

### ISSUE THREAD

**Issue by omnilogix** [microsoft/microsoft-ui-xaml#4668]
Is there an API that manipulates a WinUI window size/state from code?

I have used the TitleBar API's that I am aware of to set a custom TitleBar
```
            ExtendsContentIntoTitleBar = true;
            SetTitleBar(myControl);

```
While this works to set the TitleBar, the Minimize/Maximize/Close buttons are now unresponsive.  I would like to overlay my own buttons to restore the broken behavior. Or, if there is a better example for how to set the custom TitleBar that does not disable the main window buttons, that would be a great solution as well.  I posted this on StackOverflow, but have not had any responses.

This is for **WinUI 3 (Project Reunion 0.5), Win32 Desktop**

**Comment by StephenLPeters** [comment]
@omnilogix seems like the real issue is that the min max and close buttons are unresponsive right? @pratikone and @Austin-Lamb .

**Comment by StephenLPeters** [comment]
@MikeHillberg do you know about how to programmatically minimize a win32 app?

**Comment by omnilogix** [comment]
> @omnilogix seems like the real issue is that the min max and close buttons are unresponsive right? @pratikone and @Austin-Lamb .

Yes!  I'm not sure if I'm missing something, or this is broken...   Looking for a solid example that works

**Comment by omnilogix** [comment]
@andrewleader - per your request at the standup.  This question could be answered as "This is currently broken", or "here is a link to an updated example".   Surely someone knows if an API of a current release is completely non-functional.   While I wait for an answer, I continue to try scenarios that might work.  This obviously takes time and resources.  A timely response from MS allows me to move forward with other priorities.   We are currently prototyping our next-gen release for a very complex WPF app, and all of the delays waiting for answers really add up.  I try to post questions when the documentation is sparse instead of posting a bug report.  I think questions should be responded to swiftly.  Bugs obviously take more time to test and prioritize.  Perhaps MS would find that the backlog of bug reports could be significantly reduced by being more responsive to questions and improving documentation.  Either way, getting a response of some kind is much better than seeing that someone was "alerted"

**Comment by andrewleader** [comment]
Thanks omnilogix for bringing this up, really sorry you didn't get a timely response!

Ping @MikeHillberg, question for you above.

**Comment by marb2000** [comment]
@omnilogix The custom Title bar has a couple of bugs that depending on the XAML layout caption buttons aren't rendered on the top of the visual tree. We have bug fixes ready for 0.8 GA.  
About  Maximize, Minimize, and Restore WinUI window from code, you can find some samples at [DesktopWindow](https://github.com/marb2000/DesktopWindow) that shows you how you can achieve it using Win32 APIs. These window functionalities will be added post v1, per the current engineering plan.
